python/pyplugin_installer/installer_data.py

```diff
@@ -860,7 +860,10 @@ def pluginMetadata(fct):
                 pyQgisVersion(), qgisMinimumVersion, qgisMaximumVersion
             ):
                 error = "incompatible"
-                errorDetails = f"{qgisMinimumVersion} - {qgisMaximumVersion}"
+                errorDetails = QCoreApplication.translate(
+                    "QgsPluginInstaller",
+                    f"Plugin designed for QGIS {qgisMinimumVersion} - {qgisMaximumVersion}",
+                )
```

Comment:
```suggestion
                errorDetails = QCoreApplication.translate(
                    "QgsPluginInstaller",
                    "Plugin designed for QGIS {0} - {1}",
                ).format(qgisMinimumVersion, qgisMaximumVersion)
```

It might be easier for translators to handle string that way